pkg/api/api.go

```diff
@@ -567,25 +703,32 @@ func (s *Service) contentLengthMetricMiddleware() func(h http.Handler) http.Hand
 	}
 }
 
-// gasConfigMiddleware can be used by the APIs that allow block chain transactions to set
-// gas price and gas limit through the HTTP API headers.
-func (s *Service) gasConfigMiddleware(handlerName string) func(h http.Handler) http.Handler {
+// validateHeaderValues aka parsingAndConfigMiddleware
+// this middelware will be added to all API endpoint handlers
+func (s *Service) validateHeaderValues(handlerName string) func(h http.Handler) http.Handler {
 	return func(h http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			logger := s.logger.WithName(handlerName).Build()
-
-			headers := struct {
-				GasPrice *big.Int `map:"Gas-Price"`
-				GasLimit uint64   `map:"Gas-Limit"`
-			}{}
-			if response := s.mapStructure(r.Header, &headers); response != nil {
-				response("invalid header params", logger, w)
-				return
-			}
 			ctx := r.Context()
-			ctx = sctx.SetGasPrice(ctx, headers.GasPrice)
-			ctx = sctx.SetGasLimit(ctx, headers.GasLimit)
 
+			for _, hd := range allHeaderDescriptors {
```

Comment:
The iteration, validation, and context setting of all headers must be reflected in each endpoint described in the Open API specification, including a description of what the headers are used for. Since most endpoints do not use all of these headers, this is not the correct approach. The Open API specification should be on pair with the implementation.

Reply:
i had a discussion about this same issue with @vladopajic on Friday or Sat and thanks to him there is an approach we can use to validate that and it also helps with actually enforcing some rules for open api docs and based on this approach i also have found a solution for it will also be sharing that later today probably, maybe we can explore it slightly more and see if it sorts out all those problems we had prev cuz like an extension of this solution will also fix the research part that we were figuring out before.
I am improving the error messages to be more user/dev friendly.